app/dbConnection.js

```diff
@@ -18,58 +19,93 @@ function connect() {
+module.exports.writeRelationships = function (relationJSON) {
   connect().then(() => {
     let relationships = context.component('models').module('relationships');
     let relationshipEntities = context.component('models').module('relationshipEntities');
+    let relationshipDescriptions = context.component('models').module('relationshipDescriptions');
 
-    relationJSON.forEach((sentence) => {
-      sentence.instances.forEach((relation) => {
-        let subjectId = null;
-        let objectId = null;
-        relationshipEntities.sync().then(() => {
+    const promises = relationJSON.reduce((acc, sentence) => {
+      return acc.concat(sentence.instances.map((relation) => {
+        let subject = null;
+        let object = null;
+        let relationshipDescription = null;
+        let stem = null;
+        return relationshipEntities.sync().then(() => {
           if (relation.term1) {
             return relationshipEntities.create({
               'name': relation.term1
             });
           }
+        }).catch(err => {
+          console.err('ERROR: ' + err);
         }).then(d => {
           if (d) {
-            subjectId = d.id;
+            subject = d;
           }
           console.log(d);
           if (relation.term2) {
             return relationshipEntities.create({
               'name': relation.term2
             });
           }
+        }).catch(err => {
+          console.err('ERROR: ' + err);
         }).then(d => {
           if (d) {
-            objectId = d.id;
+            object = d;
           }
           return relationships.sync();
+        }).catch(err => {
+          console.err('ERROR: ' + err);
         }).then(() => {
+          stem = stemmer.getStemming(relation.relation);
+          return relationshipDescriptions.findOne({
+            where: {
+              relationship_name: stem
+            }
+          })
+        }).catch(err => {
+          console.err('ERROR: ' + err);
+        }).then(relationshipDescription => {
+          if (!relationshipDescription) {
+            return relationshipDescriptions.create({
+              'relationship_name': stem,
+            })
+          }
+          return relationshipDescription;
+        }).catch(err => {
+          console.err('ERROR: ' + err);
+        }).then(d => {
+          relationshipDescription = d;
           return relationships.create({
             'confidence': relation.quality,
-            'subjectId': subjectId,
-            'objectId': objectId,
             'relation': relation.relation
           });
+        }).catch(err => {
+          console.err('ERROR: ' + err);
+        }).then(relationship => {
+          relationship.setSubject(subject);
+          relationship.setObject(object);
+          relationship.setRelationshipDescription(relationshipDescription);
         });
-      });
+      }));
+    }, []);
+
+    Promise.all(promises).then(() => {
+      console.log('finished :)');
```

Comment:
Promise.all working well so far! Only the console log "finished :)" comes too early. Maybe the log order is not the one in which the promises are fulfilled?